### DISCUSSION TITLE
Install and run inginious on server

### DISCUSSION THREAD

**Discussion by mounirwail** [INGInious/INGInious#675] [Q&A] [answered]
hello!
![Screenshot from 2021-05-25 00-50-15](https://user-images.githubusercontent.com/18490889/119419960-3825ac80-bcf3-11eb-9315-4aafa9462956.png)
I am facing this problem when running a server inginious-webapp
Please help me solve the problem

**Comment by anthonygego** [comment, marked as answer]
Did you perform a system-wide install or a virtualenv install ? Which version of docker-py do you have ? In other words, what does ``pip show docker`` returns ?

**Reply by mounirwail** [reply]
I installed it on the original system
![image](https://user-images.githubusercontent.com/18490889/119486557-4a820380-bd50-11eb-8249-586438158408.jpeg)

**Reply by anthonygego** [reply]
Here you show the docker client version. By typing ``pip show docker`` you will get the information about the docker Python API. You may have an older version installed. The latest available (5.0.0) seems to be good.

You can also see if the older package is installed by typing ``pip show docker-py``. If it is installed, it will probably conflict.

In all cases, I recommend you to setup INGInious in a separate Python virtual environment to avoid any version conflict with your distribution packages.

**Reply by mounirwail** [reply]
thank you for helping me
I have installed it in a virtual environment
![image](https://user-images.githubusercontent.com/18490889/119517004-22ee6380-bd6f-11eb-906c-1b5af15f3085.jpeg)![image](https://user-images.githubusercontent.com/18490889/119517039-2a157180-bd6f-11eb-99f6-20e4cfd25dbb.png)

But the same problem

**Reply by mounirwail** [reply]
The V0.6 edition worked normally with me, but the V0.6 edition. I face the problem that is at the top when running the inginious-webapp server

**Reply by anthonygego** [reply]
That's actually not the same error. 

The web framework was changed between ``v0.6`` and ``master``. I realize the docs for v0.6 might not be up to date and should display ``pip install --upgrade git+https://github.com/UCL-INGI/INGInious@inginious-0.6`` instead of ``pip install --upgrade git+https://github.com/UCL-INGI/INGInious@v0.6``. Please let me know if that allows you to run the app. 

Also note that this branch may not work correctly on Python3.7 and above. You can use the master branch with newer Python versions.

**Reply by mounirwail** [reply]
Python 3.9.0+
The server has been successfully launched
But when entering the link:
http://localhost:8080
I see a blank page on my browser
I see errors on the command 
( 
/home/mounir/boutiba/boutiba/lib/python3.9/site-packages/inginious/frontend/templates/internalerror.html:39 - execution of 'Constant' statements is denied
/home/mounir/boutiba/boutiba/lib/python3.9/site-packages/inginious/frontend/templates/internalerror.html:39 - execution of 'Constant' statements is denied
/home/mounir/boutiba/boutiba/lib/python3.9/site-packages/inginious/frontend/templates/internalerror.html:40 - execution of 'Constant' statements is denied
/home/mounir/boutiba/boutiba/lib/python3.9/site-packages/inginious/frontend/templates/internalerror.html:40 - execution of 'Constant' statements is denied
)

**Reply by anthonygego** [reply]
These errors come from web.py, so I guess you're deploying v0.6. As I said this version won't work correctly on Python3.7 and above. Either use Python3.6, or deploy the master branch (remove ``@inginious-0.6``).
